app/src/main/java/acr/browser/lightning/settings/fragment/AdBlockSettingsFragment.kt

```diff
@@ -414,6 +415,38 @@ override fun onDestroy() {
         super.onActivityResult(requestCode, resultCode, data)
     }
 
+    // class necessary to allow separate onClickListener for the switch
+    private inner class FilterListSwitchPreference(context : Context?, val entity: AbpEntity) : SwitchPreferenceCompat(context) {
+
+        override fun onBindViewHolder(holder: PreferenceViewHolder?) {
+            super.onBindViewHolder(holder)
+            val switch: SwitchCompat? = holder?.itemView?.findViewById(R.id.filter_list_switch_widget)
```

Comment:
Nice trick. I wonder where you got that magic id from?

Reply:
It's actually not a special id, I only replace the original switch with a new one I can actually access (via `widgetLayoutResource`).
I didn't find a way of accessing the ["original" switch](https://android.googlesource.com/platform/frameworks/support/+/671d9d6/v7/preference/src/android/support/v7/preference/SwitchPreferenceCompat.java#137).